csp/middleware.py

```diff
@@ -65,6 +66,10 @@ def process_response(self, request, response):
         replace = getattr(response, '_csp_replace', None)
         nonce = getattr(request, '_csp_nonce', None)
 
+        report_percentage = getattr(settings, 'CSP_REPORT_PERCENTAGE', 1)
+        include_report_uri = random.random() < report_percentage
```

Comment:
This isn't really doing what you expect it does, it reduces the amount of reports send but it's never exactly the percentage you are configuring thus leading to false expectations imho.

This should instead use the django-cache imho and update the docs with a configuration warning similar to django-rest-framework: http://www.django-rest-framework.org/api-guide/throttling/#setting-up-the-cache

Reply:
Is it really worth the extra complexity though? The motivation for this setting is to stop spamming your `report-uri`. Do we really have to guarantee 100% accuracy here? What value does that provide?

Reply:
I do think it's worth the complexity, given that a complete random behaviour will avoid some reports to be sent at all which doesn't sound like something I'd want. Eventually, you want to take action on those CSP reports right?